text/0000-router-extensibility.md

```diff
@@ -0,0 +1,165 @@
+- Start Date: 2018-01-13
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Router DSL extensibility
+
+## Summary
+
+Adding a public hook to the Ember Router and one to Ember.Route would allow for addon authors and advanced app developers to enrich the route definition process in expressive ways that are not currently possible with existing hooks.
```

Comment:
The summary feels like I'm starting in the middle of a thought. I'm not sure if this is an improvement, but it's an attempt:

> Allow the router DSL to be extended to increase the router maps expressiveness and utility for addon authors and advanced app developers.